qutip/solver/result.py

```diff
@@ -0,0 +1,586 @@
+                    for traj in self.trajectories[:ntraj]]), axis=0)
+               for i in range(num_e)]
+
+        result = []
+        for expect_vals, isreal  in zip(avg, _e_type):
+            if isreal:
+                expect_vals = expect_vals.real
+            result.append(expect_vals)
+
+        if _e_ops_dict:
+            result = {e: result[n]
+                      for n, e in enumerate(_e_ops_dict.keys())}
+        return result
+
+    @property
+    def collapse(self):
+        return [traj.collapse for traj in self.trajectories]
+
+    @property
+    def col_times(self):
+        out = []
+        for col_ in self.collapse:
+            col = list(zip(*col_))
+            col = ([] if len(col) == 0 else col[0])
+            out.append(col)
+        return out
+
+    @property
+    def col_which(self):
+        out = []
+        for col_ in self.collapse:
+            col = list(zip(*col_))
+            col = ([] if len(col) == 0 else col[1])
+            out.append(col)
+        return out
+
+    @property
+    def photocurrent(self):
+        cols = {}
+        tlist = self.trajectories[0].times
+        for traj in self.trajectories:
+            for t, which in traj.collapse:
+                if which in cols:
+                    cols[which].append(t)
+                else:
+                    cols[which] = [t]
+        mesurement = []
+        for i in range(self.num_c_ops):
+            mesurement += [(np.histogram(cols.get(i,[]), tlist)[0]
+                          / np.diff(tlist) / len(self.trajectories))]
+        return mesurement
+
+    @property
+    def run_stats(self):
+        return self.trajectories[0].stats
+
+    def __repr__(self):
+        out = ""
+        out += self.run_stats['solver'] + "\n"
+        out += "solver : " + self.stats['method'] + "\n"
+        out += "{} runs saved\n".format(self.num_traj)
+        out += "number of expect : {}\n".format(self.trajectories[0]._e_num)
+        if self.trajectories[0]._store_states:
+            out += "Runs states saved\n"
+        elif self.trajectories[0]._store_final_state:
+            out += "Runs final state saved\n"
+        else:
+            out += "State not available\n"
+        out += "times from {} to {} in {} steps\n".format(self.times[0],
+                                                          self.times[-1],
+                                                          len(self.times))
+        return out
+
+    @property
+    def times(self):
+        return self.trajectories[0].times
+
+    @property
+    def states(self):
+        return self.average_states
+
+    @property
+    def expect(self):
+        return self.average_expect
+
+    @property
+    def final_state(self):
+        return self.average_final_state
+
+    @property
+    def num_traj(self):
+        return len(self.trajectories)
+
+
+class MultiTrajResultAveraged:
+    """
+    Contain result of simulations with multiple trajectories.
```

Comment:
I think the initialisation parameter `c_ops` should be documented in the class definition or the __init__.